src/ape/api/networks.py

```diff
@@ -283,10 +294,11 @@ class ProviderContextManager:
 
     # NOTE: Class variable, so it will manage stack across instances of this object
     _connected_providers: List["ProviderAPI"] = []
+    network_manager: "NetworkManager"
 
-    def __init__(self, network_manager: "NetworkManager", provider: "ProviderAPI"):
-        self.network_manager = network_manager
+    def __init__(self, provider: "ProviderAPI", network_manager: "NetworkManager"):
         self.provider = provider
```

Comment:
Do we still the to pass in the `Provider`? The `Provider` should already be accessible from the `NetworkManager` (such as via `active_provider`)

Maybe the flow here needs adjustments?

Reply:
This is the code that actually manages `active_provider`

Reply:
This is resolved, just leaving open for @unparalleled-js to read